From 32343738863542dde683b1128241f076bdfb9e58 Mon Sep 17 00:00:00 2001
From: Jonathan Hsieh <jmhsieh@apache.org>
Date: Tue, 17 Dec 2013 19:24:02 +0000
Subject: [PATCH 19/63] HBASE-10188 Deprecate ServerName constructors, but make it public (Nicolas Liochon)

git-svn-id: https://svn.apache.org/repos/asf/hbase/branches/0.96@1551661 13f79535-47bb-0310-9956-ffa450edef68

(cherry picked from 1ddeb88c578547f94acd6618251dc98aecbe4f51)

Reason: Backward compatibility
Author: Nicholas Liochon
Ref: CDH-21387
---
 .../java/org/apache/hadoop/hbase/ServerName.java   |   18 +++++++++++++++---
 1 files changed, 15 insertions(+), 3 deletions(-)

diff --git a/hbase-client/src/main/java/org/apache/hadoop/hbase/ServerName.java b/hbase-client/src/main/java/org/apache/hadoop/hbase/ServerName.java
index 48b4a79..5199322 100644
--- a/hbase-client/src/main/java/org/apache/hadoop/hbase/ServerName.java
+++ b/hbase-client/src/main/java/org/apache/hadoop/hbase/ServerName.java
@@ -96,7 +96,11 @@ public class ServerName implements Comparable<ServerName>, Serializable {
   private byte [] bytes;
   public static final List<ServerName> EMPTY_SERVER_LIST = new ArrayList<ServerName>(0);
 
-  private ServerName(final String hostname, final int port, final long startcode) {
+  /**
+   * @deprecated since 0.96, made private in 0.98. use {@link #valueOf(String, int, long)} instead.
+   */
+  @Deprecated
+  public ServerName(final String hostname, final int port, final long startcode) {
     // Drop the domain is there is one; no need of it in a local cluster.  With it, we get long
     // unwieldy names.
     this.hostnameOnly = hostname;
@@ -116,12 +120,20 @@ public class ServerName implements Comparable<ServerName>, Serializable {
     return parts[0];
   }
 
-  private ServerName(final String serverName) {
+  /**
+   * @deprecated since 0.96, made private in 0.98. use {@link #valueOf(String)} instead.
+   */
+  @Deprecated
+  public ServerName(final String serverName) {
     this(parseHostname(serverName), parsePort(serverName),
       parseStartcode(serverName));
   }
 
-  private ServerName(final String hostAndPort, final long startCode) {
+  /**
+   * @deprecated since 0.96, made private in 0.98+. use {@link #valueOf(String, long)} instead.
+   */
+  @Deprecated
+  public ServerName(final String hostAndPort, final long startCode) {
     this(Addressing.parseHostname(hostAndPort),
       Addressing.parsePort(hostAndPort), startCode);
   }
-- 
1.7.0.4

